### PR TITLE
Add articles resource CRUD

### PR DESCRIPTION
**Pull request by tpan**
Set up CRUD functionality and corresponding routes/views/models/controllers :tada:

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,18 +19,22 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
-      # Add or replace dependency steps here
```
Comment (aarondufall) [inline .github/workflows/test.yml:22]: Not a huge deal, but something this file should really be in its own PR

**Review by aarondufall — COMMENTED** [review]
LGTM

File: spec/requests/articles_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe "Articles", type: :request do
+  describe "GET article index" do
+    it "returns http success" do
+      get "/articles#index"
```
Comment (jensen) [inline spec/requests/articles_spec.rb:6]: I don't think this is doing what you intend it to do. A URL has a few parts..

protocol://host:port/path

We can also add a `#` to the end or a `?` to the end and then we get more behaviour.

So `/articles#index` will behave the same as `/articles`. This overlaps with the route config suggesting that there is an articles controller with an index action.

I would use `articles_path` here.

File: .github/workflows/test.yml (hunk 1 above)
Comment (tpan) [reply .github/workflows/test.yml:22]: @aarondufall Fair point, in this case however if I had merged this PR without this `yarn install` step, I would be introducing a failure to the pipeline. I think merging just this change, before the actions implemented in this PR wouldn't have installed any visible/used dependencies? :thinking:
